### PR TITLE
Least mental model breaking way to add a color scattered variable

### LINKED ISSUE #981: Rename ray scattered and color scattered

### PR DESCRIPTION
**Pull request by trevordblack**
Fixes #981

### CODE REVIEW

File: books/RayTracingTheRestOfYourLife.html [hunk 1]
```diff
@@ -3171,9 +3171,11 @@
             ray scattered = ray(rec.p, p.generate(), r.time());
             auto pdf_val = p.value(scattered.direction());
 
-            return emitted
-                + srec.attenuation * rec.mat->scattering_pdf(r, rec, scattered)
-                                   * ray_color(scattered, depth-1) / pdf_val;
+            color scattered_color = (srec.attenuation
+                * rec.mat->scattering_pdf(r, rec, scattered)
+                * ray_color(scattered, depth-1)) / pdf_val;
```
Comment (hollasch) [inline books/RayTracingTheRestOfYourLife.html:3176]: What would you think about something like the following?

```
auto sample_color = ray_color(scattered, depth-1);
auto scattering_pdf = rec.mat->scattering_pdf(r, rec, scattered);
color scattered_color = (srec.attenuation * scattering_pdf * sample_color) / pdf_val;
```

**Review by hollasch — CHANGES_REQUESTED** [review]
My one comment is more a suggestion than a request.

**Comment by trevordblack** [comment]
I went through a used a model that was I was comfortable with.

I also went through and removed a lot of inconsistencies, and spun a bug #988

**Comment by trevordblack** [comment]
@hollasch Please take a look

File: books/RayTracingTheRestOfYourLife.html (hunk 1 above)
Comment (trevordblack) [reply books/RayTracingTheRestOfYourLife.html]: Let me think about this one. I've created a separate issue #989  to think about it.

**Comment by hollasch** [comment]
Feels good, man.

**Pull request merged**